cgmes/cgmes-model/src/main/java/com/powsybl/cgmes/model/CgmesNamespace.java

```diff
@@ -21,32 +22,70 @@ private CgmesNamespace() {
-    public static final String TP_PROFILE = "http://entsoe.eu/CIM/Topology/4/1";
-    public static final String SV_PROFILE = "http://entsoe.eu/CIM/StateVariables/4/1";
-    public static final String SSH_PROFILE = "http://entsoe.eu/CIM/SteadyStateHypothesis/1/1";
+    private static final String CIM_16_EQ_PROFILE = "http://entsoe.eu/CIM/EquipmentCore/3/1";
+    private static final String CIM_16_EQ_OPERATION_PROFILE = "http://entsoe.eu/CIM/EquipmentOperation/3/1";
+    private static final String CIM_16_TP_PROFILE = "http://entsoe.eu/CIM/Topology/4/1";
+    private static final String CIM_16_SV_PROFILE = "http://entsoe.eu/CIM/StateVariables/4/1";
+    private static final String CIM_16_SSH_PROFILE = "http://entsoe.eu/CIM/SteadyStateHypothesis/1/1";
 
-    public static final Set<String> CIM_NAMESPACES = Set.of(CIM_14_NAMESPACE, CIM_16_NAMESPACE, CIM_100_NAMESPACE);
+    private static final String CIM_100_EQ_PROFILE = "http://iec.ch/TC57/ns/CIM/CoreEquipment-EU/3.0";
+    private static final String CIM_100_EQ_OPERATION_PROFILE = "http://iec.ch/TC57/ns/CIM/Operation-EU/3.0";
+    private static final String CIM_100_TP_PROFILE = "http://iec.ch/TC57/ns/CIM/Topology-EU/3.0";
+    private static final String CIM_100_SV_PROFILE = "http://iec.ch/TC57/ns/CIM/StateVariables-EU/3.0";
+    private static final String CIM_100_SSH_PROFILE = "http://iec.ch/TC57/ns/CIM/SteadyStateHypothesis-EU/3.0";
 
-    public static String getCimNamespace(int cimVersion) {
-        if (cimVersion == 14) {
-            return CIM_14_NAMESPACE;
-        }
-        if (cimVersion == 16) {
-            return CIM_16_NAMESPACE;
+    private static final Map<Integer, Map<String, String>> PROFILES = Map.of(
+            16, Map.of("EQ", CIM_16_EQ_PROFILE, "EQ_OP", CIM_16_EQ_OPERATION_PROFILE, "SSH", CIM_16_SSH_PROFILE, "SV", CIM_16_SV_PROFILE, "TP", CIM_16_TP_PROFILE),
+            100, Map.of("EQ", CIM_100_EQ_PROFILE, "EQ_OP", CIM_100_EQ_OPERATION_PROFILE, "SSH", CIM_100_SSH_PROFILE, "SV", CIM_100_SV_PROFILE, "TP", CIM_100_TP_PROFILE)
+    );
+
+    public static boolean isValid(String ns) {
+        // Until CIM16 the CIM namespace contained the string "CIM-schema-cim<versionNumber>#"
+        // Since CIM100 the namespace seems to follow the pattern "/CIM<versionNumber>#"
+        return VALID_CIM_NAMESPACES.contains(ns) || CIM_100_PLUS_NAMESPACE_PATTERN.matcher(ns).matches();
+    }
+
+    public static boolean isValidAny(Set<String> namespaces) {
+        // FIXME(Luma) This is legacy behaviour, we do not consider CIM14 valid in this check
+        // But I think we do not need to support 14 separately?
+        return namespaces.contains(CIM_16_NAMESPACE) || namespaces.contains(CIM_100_NAMESPACE);
+    }
+
+    public static boolean isValidAny14(Set<String> namespaces) {
+        return namespaces.contains(CIM_14_NAMESPACE);
+    }
+
+    public static String getCim(int cimVersion) {
```

Comment:
Why did you rename this method?

Reply:
It seemed redundant. It is a static method, and the class name already talks about namespace, so the calls will look like: `CgmesNamespace.getCim(cimVersion);`